src/plugins/data_view_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -99,10 +132,39 @@ export const IndexPatternTable = ({
         gettedIndexPatterns.length === 0 ||
         !(await dataViews.hasUserDataView().catch(() => false))
       ) {
-        setShowCreateDialog(true);
+        setHasIndexPatterns(false);
+      }
+      getIndices({
+        http,
+        isRollupIndex: () => false,
+        pattern: '*',
+        showAllIndices: false,
+        searchClient: data.search.search,
+      }).then((dataSources) => {
+        setHasDataIndices(dataSources.some(isUserDataIndex));
+      });
+      if (!hasDataIndices && !hasCheckedRemoteClusters) {
```

Comment:
@mattkime  Are these checks okay? I replicated them from `empty_prompts.tsx` and `src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx` to check whether data exists or not.